src/shared/redux/slices/loadPublicationsSlice.ts

```diff
@@ -5,29 +5,17 @@ import { RootState } from '../rootReducer';
 // type created to avoid an issue with both unclear modification type
 // and index access writes on union of keys, which would complicate the
 // code significantly more
-type PublicationModification =
-  | {
-      id: string;
-      field: keyof Omit<
-        Publication,
-        | 'useTypescript'
-        | 'useSass'
-        | 'collaborators'
-        | 'creationDate'
-        | 'status'
-      >;
-      value: string;
-    }
-  | {
-      id: string;
-      field: keyof Pick<Publication, 'useTypescript' | 'useSass'>;
-      value: boolean;
-    }
-  | {
+
+type PublicationModification = Exclude<
+  {
+    [FieldName in keyof Publication]: {
       id: string;
-      field: keyof Pick<Publication, 'status'>;
-      value: 'cloned' | 'remote';
+      field: FieldName;
+      value: Publication[FieldName];
```

Comment:
If we'd want the array values to be the type of the array element we could do the following as well. But I don't see that really coming in handy.

```ts
value: Publication[FieldName] extends Array<any>
        ? Publication[FieldName][0]
        : Publication[FieldName];
```

The most versatile thing for array properties would be having `add`, `remove` and `set` actions but you can see ass you go if they are needed.